techtonica-projects/animal-sighting-tracker/server/server.js

```diff
@@ -0,0 +1,329 @@
+const express = require('express');
+const cors = require('cors');
+require('dotenv').config();
+const path = require('path');
+const db = require('./db/db-connection.js');
+
+
+const app = express();
+const PORT = process.env.PORT || 8080;
+app.use(cors());
+app.use(express.json());
+app.use('/images', express.static('public/images'));
+
+// creates an endpoint for the route "/""
+app.get('/', (req, res) => {
+    res.json({ message: 'Hola, from My template ExpressJS with React-Vite' });
+});
+
+// ----------------------------CRUD for 'species' table----------------------------------------------//
+
+//CREATE species
+// app.post('/api/species', async (req, res) => {
+//     try {
+//         const newSpecies = {
+//             common_name: req.body.common_name,
+//             scientific_name: req.body.scientific_name,
+//             number_in_wild: req.body.number_in_wild,
+//             csc: req.body.conservation_status_code
+//         };
+       
+//         const result = await db.query(
+//             'INSERT INTO species(common_name, scientific_name, number_in_wild, conservation_status_code) VALUES($1, $2, $3, $4) RETURNING *',
+//             [newSpecies.common_name, newSpecies.scientific_name, newSpecies.number_in_wild, newSpecies.conservation_status_code],
+//         );
+//         console.log(result.rows[0]);
+//         res.json(result.rows[0]);
+
+//     } catch (e) {
+//         console.log(e);
+//         return res.status(400).json({ e });
+//     }
+
+// });
+
+//READ species
+app.get('/api/species', async (req, res) => {
+    try {
+        const { rows: species } = await db.query('SELECT * FROM species');
+        res.send(species);
+    } catch (e) {
+        return res.status(400).json({ e });
+    }
+});
+
+//READ species JOIN query (shows a count for how many individuals are tracked)
```

Comment:
Are these commented out so they can be used at a later time? 
If so, I'd just add a TODO at the top!

Reply:
Yes, good idea.